### PR TITLE
Mfreshour.alb create

### PR DESCRIPTION
**Pull request by mgfreshour**

### CODE REVIEW

File: src/cli/arg_handler.js [hunk 1]
```diff
@@ -31,6 +31,24 @@ function _handle_create(cmd) {
...
 			break;
 
+		case 'alb':
+			nemesys.alb.create(
+				cmd.opts['regions'],
+				cmd.opts['vpc'],
+				cmd.opts['security-group'],
+				cmd.opts['name'],
+				cmd.opts['target_groups'],
+				cmd.opts['ssl_config'],
```
Comment (torra) [inline src/cli/arg_handler.js]: can we hyphenate to be consistent w/ other param names?

File: src/cli/parse_args.js [hunk 2]
```diff
@@ -134,6 +134,20 @@ function parse_args (args) {
...
+
+						.option('v', vpc_opt)
+						.option('name', {
+							alias: 'n',
+							describe: 'Name of ALB',
+						})
+						.option('security-group', {
+							alias: 's',
+							describe: 'Security Group name new ALB should be tied to.'
+						})
```
Comment (torra) [inline src/cli/parse_args.js]: is this missing the 'ssl config', 'target group', and 'options' options?

File: src/api/aws_provider.js [hunk 3]
```diff
@@ -38,9 +38,9 @@ function _get_cloudwatch(region) {
 
 function _get_elb(region) {
 	if(!elb_conns[region]) {
-		elb_conns[region] = BB.promisifyAll(new AWS.ELB({
+		elb_conns[region] = BB.promisifyAll(new AWS.ELBv2({
```
Comment (torra) [inline src/api/aws_provider.js]: im hoping there are no breaking changes here... but im not sure what the best way to catch this kind of thing would be. better tests i guess? the problem is that all of the tests mock out the api anyways, so if there is a breaking change in the response format, we wouldnt know until we try to run it for real

File: tst/api/alb/create.test.js [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+	});
+
+	it('creates LB without targets', function () {
+		return testee(regions, vpc_name, sg_name, lb_name, target_groups, ssl_config, options)
+			.then(() => {
+				expect(mock_elb.createLoadBalancerAsync.callCount).to.equal(1);
+				expect(mock_elb.createTargetGroupAsync.callCount).to.equal(1);
+				expect(mock_elb.createListenerAsync.callCount).to.equal(1);
+				expect(mock_elb.registerTargetsAsync.callCount).to.equal(0);
```
Comment (torra) [inline tst/api/alb/create.test.js]: do you think it would be helpful for the tests to check that the aws api calls have the expected arguments?

File: src/api/aws_provider.js (hunk 3 above)
Comment (mgfreshour) [inline src/api/aws_provider.js]: Ah, yeah, I should just make a get_elb2 function.  When I search for usage, I searched for the function name, I didn't notice at the time it is renamed on export.

File: tst/api/alb/create.test.js (hunk 4 above)
Comment (mgfreshour) [inline tst/api/alb/create.test.js]: Yeah, I plan on making the tests better in a future PR.  This one was just getting really big, so I wanted to get it out before adding 200+ more lines :)

File: src/cli/parse_args.js (hunk 2 above)
Comment (mgfreshour) [inline src/cli/parse_args.js]: Since they are JSON constructs, I wasn't for sure how to add them to command line options.

File: src/cli/arg_handler.js (hunk 1 above)
Comment (mgfreshour) [inline src/cli/arg_handler.js]: Will do!

File: src/cli/parse_args.js (hunk 2 above)
Comment (torra) [inline src/cli/parse_args.js]: hmm yea, ive had the same issue and im not really sure the best way to handle it. what ive done in other places is to use some kind of formatted string, but for nested objects that would get tricky. maybe we could just accept like a `JSON.stringify()`'d object (i think aws cli does that for some things)

File: tst/api/alb/create.test.js (hunk 4 above)
Comment (torra) [inline tst/api/alb/create.test.js]: sounds good

File: src/cli/parse_args.js [hunk 5]
```diff
@@ -100,7 +100,7 @@ function parse_args (args) {
 				.alias('h', 'help');
 		})
 
-		.command('create', 'Create an EC2 resource', function (yargs) {
+		.command('create', 'Create an AWS resource', function (yargs) {
```
Comment (torra) [inline src/cli/parse_args.js]: just curious, why the change? this is still EC2

File: src/api/alb/create.js [hunk 6]
```diff
@@ -0,0 +1,221 @@
+'use strict';
+
+const BB = require('bluebird');
+
+const Logger      = require('../../logger');
+const AWSUtil     = require('../aws_util');
+const AWSProvider = require('../aws_provider');
+
+function _create_elb (aws, lb_name, subnet_ids, opts, sg_id) {
```
Comment (torra) [inline src/api/alb/create.js]: should be `_create_alb()`?

File: src/cli/parse_args.js (hunk 5 above)
Comment (mgfreshour) [inline src/cli/parse_args.js]: :shrug: I have no clue.  Maybe I was feeling unusually pedantic or something.
